ui/src/source.ts

```diff
@@ -112,15 +117,24 @@ export enum RevisionType {
   Sha = "sha",
 }
 
+// Client representation of a git branch
 export interface Branch {
   type: RevisionType.Branch;
   name: string;
-  peerId?: string;
+  // The id of the project this branch belongs to
+  projectId: string;
```

Comment:
```suggestion
  projectUrn: string;
```

Reply:
This is worthy of its own pr because `Project` has an `id` in proxy - relevant to #840